store/sheet_organizer.go

```diff
@@ -0,0 +1,91 @@
+package store
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/bytebase/bytebase/api"
+)
+
+// sheetOrganizerRaw is the store model for SheetOrganizer.
+type sheetOrganizerRaw struct {
+	ID int
+
+	// Related fields
+	SheetID     int
+	PrincipalID int
+	Starred     bool
+	Pinned      bool
+}
+
+// toSheetOrganizer creates an instance of SheetOrganizer based on the sheetOrganizerRaw.
+func (raw *sheetOrganizerRaw) toSheetOrganizer() *api.SheetOrganizer {
+	return &api.SheetOrganizer{
+		ID:          raw.ID,
+		SheetID:     raw.SheetID,
+		PrincipalID: raw.PrincipalID,
+		Starred:     raw.Starred,
+		Pinned:      raw.Pinned,
+	}
+}
+
+// UpsertSheetOrganizer upserts a new SheetOrganizer.
+func (s *Store) UpsertSheetOrganizer(ctx context.Context, upsert *api.SheetOrganizerUpsert) (*api.SheetOrganizer, error) {
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, FormatError(err)
+	}
+	defer tx.PTx.Rollback()
+
+	sheetOrganizerRaw, err := upsertSheetOrganizer(ctx, tx.PTx, upsert)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := tx.PTx.Commit(); err != nil {
+		return nil, FormatError(err)
+	}
+
+	sheetOrganizer := sheetOrganizerRaw.toSheetOrganizer()
+
+	return sheetOrganizer, nil
+}
+
+func upsertSheetOrganizer(ctx context.Context, tx *sql.Tx, upsert *api.SheetOrganizerUpsert) (*sheetOrganizerRaw, error) {
```

Comment:
```suggestion
func upsertSheetOrganizerImpl(ctx context.Context, tx *sql.Tx, upsert *api.SheetOrganizerUpsert) (*sheetOrganizerRaw, error) {
```

To be consistent with other implementations like `Principal`.